plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -774,6 +774,30 @@ public void testLimitPushdown()
                 .isFullyPushedDown();
     }
 
+    @Override
+    public void testCaseSensitiveTopNPushdown()
+    {
+        // TODO: Remove this override once SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY is true (https://github.com/trinodb/trino/issues/7496)
```

Comment:
Or we may want to model this as we did with Joins?

see `SUPPORTS_JOIN_PUSHDOWN_WITH_VARCHAR_*EQUALITY`

Reply:
If we can figure out a theoretical solution to allow connectors to participate in predicate pushdown query building I'd keep the override - too many similar enums will confuse implementers (and reviewers) and will make it more work to cleanup in future.

If we don't have a way to also fix predicate pushdown similar to how I fixed topn here then I'll add those new enums as you suggested.

Reply:
i don't have a short-term goal to make QueryBuilder more elastic, so it seems it's eiehter "too many enum options" or "a test override" for interim... of the two, it seems "too many enum options" is easier to maintain (and we already have the case with joins)

Reply:
picking the default value for the new option is key here -- if you do this right, only PostgreSQL tests will care about the change.